egs/wsj/s5/steps/nnet3/train_dnn.py

```diff
@@ -375,7 +375,28 @@ def TrainOneIteration(dir, iter, srand, egs_dir,
         do_average = True
         if iter == 0:
             do_average = False   # on iteration 0, pick the best, don't average.
-        raw_model_string = "nnet3-am-copy --raw=true --learning-rate={0} {1}/{2}.mdl - |".format(learning_rate, dir, iter)
+        
+        if add_ephemeral_connection:
+          if dropout_proportion == 1.0 and not os.path.exists('{0}/configs/dropout_removed'.format(dir)):
+            # If dropout_proportion is equal to 1.0, 
+            # we replace Tdnn_{0}_relu input from sum(affine, ephemeral_affine) to affine.
+            # and remove orphan nodes and components using edits-config.
+            replace_config_str = '{0}/configs/dropout.config'.format(dir)
+            replace_config_file = open(replace_config_str, 'w')
+            for layer in range(1, num_hidden_layers):
```

Comment:
Can you explain what the difference between num_hidden_layers and num_real_hidden_layers is?
I don't like the name 'num_real_hidden_layers' much.  What is this used for?